src/solver/signal-handling/common.cpp

```diff
@@ -4,29 +4,38 @@
 
 using namespace Antares;
 
-static void finalizeWrite()
+namespace Antares::Solver {
+
+using Antares::Solver::IResultWriter;
+
+// weak_ptr because the lifetime must not be managed by this feature.
+// If the study lifetime ends before we receive a signal, we shoud just do
+// nothing when receiving the signal.
+static std::weak_ptr<IResultWriter> APPLICATION_WRITER;
+
+void setApplicationResultWriter(std::weak_ptr<IResultWriter> writer)
 {
-    auto study = Data::Study::Current::Get();
-    if (study)
-    {
-        auto writer = study->resultWriter;
-        if (writer)
-            writer->finalize(true);
-        else
-        {
-            logs.warning() << "Could not finalize write: invalid writer";
-            exit(EXIT_FAILURE);
-        }
-    }
-    else
-    {
-        logs.warning() << "Could not finalize write: invalid study";
+    APPLICATION_WRITER = writer;
+}
+
+}
+
+namespace {
+
+static void finalizeWrite() {
```

Comment:
```suggestion
void finalizeWrite() {
```

Everything is `static` in the anonymous namespace.